### PR TITLE
add support for aws Cognito jwt 

### PR DESCRIPTION
**Pull request by advancedtw**
fix to support Cognito tokens. the main differences:

1. issuer url requires a path referencing the pool id 
2. the header doesn't have a type field.

### CODE REVIEW

File: packages/jwt/src/parse.ts [hunk 1]
```diff
@@ -16,12 +16,6 @@ export async function parseJwt(
   } catch {
     return { valid: false, reason: `Unable to decode JWT.` };
   }
-  if (decoded.header.typ !== 'JWT') {
```
Comment (jdanyow) [inline packages/jwt/src/parse.ts:19]: Let's continue to validate typ when it's provided.

File: packages/jwt/src/parse.ts [hunk 2]
```diff
@@ -34,8 +28,8 @@ export async function parseJwt(
       reason: `Invalid JWT audience "${decoded.payload.aud}". Expected "${audience}".`
     };
   }
-  const iss = new URL(decoded.payload.iss);
-  if (iss.origin !== issuerOrigin) {
+
+  if (decoded.payload.iss !== issuerOrigin) {
```
Comment (jdanyow) [inline packages/jwt/src/parse.ts:32]: issuerOrigin should be renamed to issuer to match the new logic

File: packages/jwt/test/parse.spec.ts [hunk 3]
```diff
@@ -30,15 +30,6 @@ describe('parseJwt', () => {
     expect(result.valid).to.equal(false);
   });
 
-  it('rejects unexpected type', async () => {
```
Comment (jdanyow) [inline packages/jwt/test/parse.spec.ts:33]: Let's keep and add another for cases when typ is not provided

**Review by jdanyow — CHANGES_REQUESTED** [review]
Thanks for the pr, few small changes and I'll merge and ship

**Comment by advancedtw** [comment]
I made the changes you requested. in addition I had to make JwtHeader.typ optional.  let me know what you think

**Review by jdanyow — APPROVED** [review]

**Comment by jdanyow** [comment]
shipped in [@cfworker/jwt 1.6.9](https://www.npmjs.com/package/@cfworker/jwt/v/1.6.9)

**Pull request merged**